app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt

```diff
@@ -0,0 +1,171 @@
+package org.jellyfin.mobile.bridge
+
+import android.app.Activity
+import android.content.Intent
+import android.webkit.JavascriptInterface
+import org.jellyfin.mobile.MainActivity
+import org.jellyfin.mobile.R
+import org.jellyfin.mobile.player.source.JellyfinMediaSource
+import org.jellyfin.mobile.utils.Constants
+import org.json.JSONException
+import org.json.JSONObject
+import org.koin.core.KoinComponent
+import timber.log.Timber
+
+class ExternalPlayer(private val activity: MainActivity) : KoinComponent {
+
+    private var mediaSource: JellyfinMediaSource? = null
+    private var playerIntent: Intent? = null
+
+    @JavascriptInterface
+    fun initPlayer(args: String) {
+        try {
+            mediaSource = JellyfinMediaSource(JSONObject(args))
+            if (mediaSource?.playMethod.equals("DirectStream")) {
+                playerIntent = Intent(Intent.ACTION_VIEW).apply {
+                    setDataAndType(mediaSource?.uri, mediaSource?.mimeType)
+                    putExtra("title", mediaSource?.title)
+                    putExtra("position", mediaSource?.mediaStartMs)
+                    putExtra("return_result", true)
+                    putExtra("secure_uri", true)
+                }
+                activity.startActivityForResult(playerIntent, Constants.HANDLE_EXTERNAL_PLAYER)
+                Timber.d("Starting playback [id: ${mediaSource?.id}, title: ${mediaSource?.title}, playMethod: ${mediaSource?.playMethod}, mediaStartMs: ${mediaSource?.mediaStartMs}, mediaDurationMs: ${mediaSource?.mediaDurationMs}]")
+            } else {
+                Timber.d("Play Method '${mediaSource?.playMethod}' not tested, ignoring...")
+                notifyEvent(
+                    Player.EVENT_CANCELED,
+                    "'${activity.getString(R.string.external_player_invalid_play_method)}'"
+                )
+            }
+        } catch (e: JSONException) {
+            Timber.e(e)
+        }
+    }
+
+    private fun notifyEvent(event: String, parameters: String = "") {
+        activity.loadUrl("javascript:window.ExtPlayer.notify$event($parameters)")
+    }
+
+    private object Player {
+        const val EVENT_ENDED = "Ended"
+        const val EVENT_TIME_UPDATE = "TimeUpdate"
+        const val EVENT_CANCELED = "Canceled"
+    }
```

Comment:
Could you move those to the Constants file please? Below the result actions for example, with a comment like `// External player webapp events`.